conftest.py

```diff
@@ -54,6 +54,21 @@ def is_single_card_n300(device):
     return num_pcie == 1 and num_devices == 2 and device.arch().name == "WORMHOLE_B0"
 
 
+def is_tg_cluster():
+    import ttnn
+
+    num_pcie = ttnn.GetNumPCIeDevices()
+    num_devices = ttnn.GetNumAvailableDevices()
+    # TG has 32 available chips and 4 PCIe mapped chips (not exposed to the user)
+    return num_pcie == 4 and num_devices == 32
```

Comment:
Maybe superficial right now, but these constants could be moved elsewhere one day.

Also, what happens if the cluster is tg, but its missing one device due to a system error?

```suggestion
    # TG has 32 available chips and 4 PCIe mapped chips (not exposed to the user)
    TG_NUM_PCIE_DEVICES = 4
    TG_NUM_DEVICES = 32
    return num_pcie ==  TG_NUM_PCIE_DEVICES and num_devices == TG_NUM_DEVICES
```

Reply:
Will use proper constant names here. 
In the case where one of the devices is missing due to a system issue, we will end up using N150 (device 0) as the single device. This will just cause the test to error out since it won't have enough worker cores on the N150. Having said that, we have a other fixtures/helper functions relying on the correct number of devices (ex: `is_single_card_n300`.
Usually in the case of a system error like the one you described, cluster descriptor generation fails with an `Unreachable chip` error before the test runs.